### PR TITLE
Allow Judges to see all information, regardless of room settings

### PR DESCRIPTION
**Pull request by ZeldaZach**

### CODE REVIEW

File: common/server_game.cpp [hunk 1]
```diff
@@ -287,18 +287,19 @@ void Server_Game::sendGameStateToPlayers()
...
-            gec = prepareGameEvent(spectatorEvent, -1);
-        else {
+        if (player->getSpectator()) {
+            if (spectatorsSeeEverything || player->getJudge()) {
+                gec = prepareGameEvent(omniscientEvent, -1);
+            } else {
+                gec = prepareGameEvent(spectatorNormalEvent, -1);
+            }
+        } else {
             Event_GameStateChanged event;
```
Comment (ebbit1q) [inline common/server_game.cpp:303]: I like the way this is cleaner now, for consistency could we move this declaration to the place the other one is?

Comment (ebbit1q) [reply common/server_game.cpp:303]: actually no, this event is unique to the player isn't it?

File: common/server_response_containers.cpp [hunk 2]
```diff
@@ -41,8 +41,10 @@ void GameEventStorage::setGameEventContext(const ::google::protobuf::Message &_g
 void GameEventStorage::enqueueGameEvent(const ::google::protobuf::Message &event,
                                         int playerId,
                                         GameEventStorageItem::EventRecipients recipients,
-                                        int _privatePlayerId)
+                                        int _privatePlayerId,
+                                        bool _overwriteOwnership)
```
Comment (ebbit1q) [inline common/server_response_containers.cpp:45]: on looking at the way setForcedByJudge is implemented I'd like to see this implemented consistently, please remove _overwriteOwnership from the enqueGameEvent function and instead have cmdRevealCards use ges.setOverwriteOwnership

**Review by ebbit1q — APPROVED** [review]
very nice, please merge this after addressing my one remark (moving one line of code)

I think we should try to make a beta with this as soon as possible.

File: common/server_game.cpp (hunk 1 above)
Comment (ZeldaZach) [reply common/server_game.cpp:303]: Correct this is unique to the player

**Pull request merged**